File: 1. Fundamentals/2DArray.py
# Initializing 2D array
N = 3
# This will Not work because the list is pass by reference. Each element will refer to the same list.
wrong_matrix = [[0 for i in range(0, N)]] * N
wrong_matrix[0][1] = 1
print(wrong_matrix)

# This is how you can initialize a 2D array.
right_matrix = [[0 for i in range(0, N)] for J in range(0, N)]
right_matrix[0][1] = 1
print(right_matrix)

File: 1. Fundamentals/2DArray.js
let N = 3;
// This will Not work because the list is pass by reference. Each element will refer to the same list.
let wrongArray = Array(N).fill(Array(N).fill(0));
wrongArray[0][1] = 1;
console.log(wrongArray);

let rightArray = Array(N)
  .fill(0)
  .map((x) => Array(N).fill(0));
rightArray[0][1] = 1;
console.log(rightArray);
